lib/soapy_cake/client.rb

```diff
@@ -63,9 +65,22 @@ def response_body(request)
     def http_response(request)
       logger.info("soapy_cake:request #{request}") if logger
 
-      url = "https://#{domain}#{request.path}"
-      HTTParty.post(url, headers: HEADERS, body: request.xml, timeout: NET_TIMEOUT).tap do |res|
-        raise RequestFailed, "Request failed with HTTP #{res.code}: #{res.body}" unless res.success?
+      uri = URI::HTTPS.build(host: domain, path: request.path)
```

Comment:
Do we need to build an URI object at all? `Net::HTTP::Post.new` (also) takes a path as its first argument.